bin/utest/util.c

```diff
@@ -1,10 +1,45 @@
 #include <stdio.h>
 #include <assert.h>
 #include <signal.h>
+#include <string.h>
+#include <setjmp.h>
 #include <stdlib.h>
 #include <sys/wait.h>
 #include <sys/fcntl.h>
 
+jmp_buf signal_return;
+static int _handled_signo;
+static siginfo_t *_save_info;
+static sigaction_t _old_sa;
+
+static void signal_handler(int signo, siginfo_t *info, void *uctx) {
+  memcpy(_save_info, info, sizeof(siginfo_t));
+  siglongjmp(signal_return, 1);
+}
+
+void setup_handler(int signo, siginfo_t *siginfo_ptr) {
+  sigaction_t sa;
+  _handled_signo = signo;
+  memset(&sa, 0, sizeof(sigaction_t));
+  memset(siginfo_ptr, 0, sizeof(siginfo_t));
+  sa.sa_sigaction = signal_handler;
+  sa.sa_flags = SA_SIGINFO;
+  int err = sigaction(_handled_signo, &sa, &_old_sa);
+  assert(err == 0);
+  _save_info = siginfo_ptr;
+}
+
+void check_signal(int signo, void *addr, int code) {
+  assert(_save_info->si_signo == signo);
```

Comment:
Since `assert` is defined as macro and uses `__FILE__` and `__LINE__` to report when the assertion failed, all messages originating from `check_signal` will point to `util.c` lines 33-35. That would makes those diagnostic messages quite useless.